src/TLuaInterpreter.cpp

```diff
@@ -371,40 +372,128 @@ int TLuaInterpreter::denyCurrentSend( lua_State * L )
-            pE.mArgumentTypeList.append( ARGUMENT_TYPE_STRING );
+        else if( lua_isstring( L, i ) ) {
+            event.mArgumentList.append( QString::fromUtf8( lua_tostring( L, i ) ) );
+            event.mArgumentTypeList.append( ARGUMENT_TYPE_STRING );
         }
-        else if( lua_isboolean( L, i ) )
-        {
-            pE.mArgumentList.append( QString::number(lua_toboolean( L, i )) );
-            pE.mArgumentTypeList.append( ARGUMENT_TYPE_BOOLEAN );
+        else if( lua_isboolean( L, i ) ) {
+            event.mArgumentList.append( QString::number( lua_toboolean( L, i ) ) );
+            event.mArgumentTypeList.append( ARGUMENT_TYPE_BOOLEAN );
         }
-        else if( lua_isnil( L, i ) )
-        {
-            pE.mArgumentList.append( QString() );
-            pE.mArgumentTypeList.append( ARGUMENT_TYPE_NIL );
+        else if( lua_isnil( L, i ) ) {
+            event.mArgumentList.append( QString() );
+            event.mArgumentTypeList.append( ARGUMENT_TYPE_NIL );
         }
-        else
-        {
-            lua_pushstring( L, tr("raiseEvent: bad argument #%1 type (expected string, number, boolean, or nil, got %2)").arg(QString::number(i), luaL_typename(L,i)).toUtf8().constData() );
+        else {
+            lua_pushstring( L, tr( "raiseEvent: bad argument #%1 type (string, number, boolean, or nil\n"
+                                   "expected, got a %2!)" )
+                            .arg( i )
+                            .arg( luaL_typename( L, i ) )
+                            .toUtf8().constData() );
             lua_error( L );
             return 1;
         }
     }
+
+    pHost->raiseEvent( event );
+
+    return 0;
+}
+
+int TLuaInterpreter::getProfileName( lua_State * L )
+{
+    Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
+    if( ! pHost ) {
+        lua_pushnil( L );
+        lua_pushstring( L, tr( "getProfileName: ERROR: NULL Host pointer!" ).toUtf8().constData() );
+        return 2;
+    }
+
+    lua_pushstring( L, pHost->getName().toUtf8().constData() );
+    return 1;
+}
+
+// raiseGlobalEvent( "eventName", ...optional arguments... )
+// sends an event to OTHER but not THIS profile {for internal events use
+// raiseEvent(...) instead!}
+// eventName is mandatory and should be a string though could be what further
+// arguments can be, i.e. strings, numbers, booleans or nils.
+int TLuaInterpreter::raiseGlobalEvent( lua_State * L )
+{
     Host * pHost = TLuaInterpreter::luaInterpreterMap[L];
-    pHost->raiseEvent( pE );
+    if( ! pHost ) {
+        lua_pushstring( L, tr( "raiseGlobalEvent: NULL Host pointer - something is wrong!" )
+                        .toUtf8().constData() );
+        lua_error( L );
+        return 1;
+    }
+
+    int n = lua_gettop( L );
+    if( ! n ) {
+        lua_pushnil( L );
+        lua_pushstring( L, tr( "raiseGlobalEvent: missing argument #1 (eventName as, probably, a string expected!)" )
```

Comment:
Um, should I instead make this a `lua_error( L )` case - it currently means that supplying NO arguments is the only _nil + error-message_ situation but, given that now there is no return value under normal cases it is largely pointless NOT to make this an error as well.